tests/integration/test_data/rubygems_packages.yaml

```diff
@@ -0,0 +1,137 @@
+# Test data for RubyGems packages
+# RubyGems package with Gemfile and Gemfile.lock containing no dependency
+# repo: The URL for the upstream git repository
+# ref: A git reference at the given git repository
+# expected_files: Expected source files <relative_path>: <file_URL>
+# expected_deps_files: Expected dependencies files (empty)
+# response_expectations: Parts of the Cachito response to check
+# content_manifest: PURLs for image contents part
+without_deps:
+  repo: https://github.com/cachito-testing/cachito-rubygems-without-deps.git
+  ref: 4f1ca077899d42064d097dad90f3565fbf4e1ce7
+  pkg_managers: ["rubygems"]
+  expected_files:
+    app: https://github.com/cachito-testing/cachito-rubygems-without-deps/tarball/4f1ca077899d42064d097dad90f3565fbf4e1ce7
+    deps/rubygems: null
+  response_expectations:
+    dependencies: []
+    packages:
+      - dependencies: []
+        name: "cachito-rubygems-without-deps"
+        type: "rubygems"
+        version: "4f1ca077899d42064d097dad90f3565fbf4e1ce7"
+  content_manifest:
+  - purl: "pkg:github/cachito-testing/cachito-rubygems-without-deps@4f1ca077899d42064d097dad90f3565fbf4e1ce7"
+# RubyGems package with GEM, GIT and PATH dependencies in Gemfile and Gemfile.lock
+# repo: The URL for the upstream git repository
+# ref: A git reference at the given git repository
+# expected_files: Expected source files <relative_path>: <file_URL>
+# expected_deps_files: Expected dependencies files <relative_path>
+# response_expectations: Parts of the Cachito response to check
+# content_manifest: PURLs for image contents part
+with_deps:
+  repo: https://github.com/cachito-testing/cachito-rubygems-with-dependencies.git
+  ref: ef6b2d7f9e4719be204531639a3ab90cfd731f26
+  pkg_managers: ["rubygems"]
+  expected_files:
+    app: https://github.com/cachito-testing/cachito-rubygems-with-dependencies/tarball/ef6b2d7f9e4719be204531639a3ab90cfd731f26
+    deps/rubygems/zeitwerk/zeitwerk-2.6.0.gem: https://rubygems.org/downloads/zeitwerk-2.6.0.gem
+    deps/rubygems/github.com/RadPad/swagger-ui_rails/swagger-ui_rails-external-gitcommit-7234e21e621b628d6b43194f9ba5cce5ca587f16/app: https://github.com/RadPad/swagger-ui_rails/tarball/7234e21e621b628d6b43194f9ba5cce5ca587f16
```

Comment:
This might just be lack of cachito understanding on my part, but aren't deps usually archive files?

Reply:
Usually yes, but for ruby the GIT dependencies need to be directories, bundler does not support archives.

Technically we never said what the format of `deps/` would be so we're not really breaking any promises :smile: 

We will need to check what it does with source containers though, I'll file a story for that

Reply:
Actually nevermind, it won't hurt source containers. OSBS just dumps the entire `remote-source.tar.gz` in there, `app/` `deps/` and everything.

Denyilst exclusion should also works, OSBS makes no distinction between files and directories (as long as the pattern matches)